Add NusaParagraph Topic Classification

gentaiscool opened the pull request:
NusaParagraphTopicClassification is a multi-class topic classification on 10 Indonesian languages.

## Checklist
<!-- Please do not delete this -->

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 


### Adding datasets checklist
<!-- see also https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md -->

**Reason for dataset addition**: ... <!-- Add reason for adding dataset here. E.g. it covers task/language/domain previously not covered -->

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

cc @Muennighoff @KennethEnevoldsen

KennethEnevoldsen commented on mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py:
```diff
@@ -0,0 +1,60 @@
+        }
+        """,
+        n_samples={"train": 15516, "validation": 2948, "test": 6250},
+        avg_character_length={"train": 5.51, "validation": 5.49, "test": 5.45},
```
again sizes seems quite small

gentaiscool replied:
let me update the sizes (previously I put the avg number of char per word)

KennethEnevoldsen commented on mteb/tasks/Classification/multilingual/NusaParagraphTopicClassification.py:
```diff
@@ -0,0 +1,60 @@
+        reference="https://github.com/IndoNLP/nusa-writes",
+        category="s2s",
+        type="Classification",
+        eval_splits=["train", "validation", "test"],
```
```suggestion
        eval_splits=["test"],
```
seems like we can test with just the test set.

gentaiscool commented:
Thank you for the review. I addressed them. I updated the avg number of character per sample, added the results, and points. I calculated the points, so there are 4 new languages `bew-Latn`, `mak-Latn`, `mui-Latn`, `rej-Latn` (2 + 4 x 4 = 18). Plus, I added the point for the reviewer. @KennethEnevoldsen
